SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -108,6 +107,21 @@ void UGlobalStateManager::ApplyDeploymentMapUpdate(const Worker_ComponentUpdate&
 		bool bUpdateAcceptingPlayers = GetBoolFromSchema(ComponentObject, SpatialConstants::DEPLOYMENT_MAP_ACCEPTING_PLAYERS_ID);
 		ApplyAcceptingPlayersUpdate(bUpdateAcceptingPlayers);
 	}
+
+	if (Schema_GetObjectCount(ComponentObject, SpatialConstants::DEPLOYMENT_MAP_SESSION_ID) == 1)
+	{
+		ApplySessionIdUpdate(Schema_GetInt32(ComponentObject, SpatialConstants::DEPLOYMENT_MAP_SESSION_ID));
+	}
+}
+
+void UGlobalStateManager::ApplySessionIdUpdate(int32 InSessionId)
+{
+	if (SessionId != InSessionId)
+	{
+		UE_LOG(LogGlobalStateManager, Log, TEXT("GlobalStateManager Update - SessionId: %lld"), InSessionId);
+		SessionId = InSessionId;
+		NetDriver->OnGSMUpdated();
```

Comment:
So this is getting triggered in both AcceptingPlayersUpdate and ApplySessionIdUpdate. We should do it in one place incase you get the updates at the same time.